Conjur/CONTRIBUTING.md

```diff
@@ -118,7 +151,233 @@ important to follow when developing.
+we make finding relevant code and automating releases easy and simple. Where possible, we
+also use GPG signing of those tags though this process has not been yet fully designed.
+
+To create an annotated tag you have to:
+- **Check out the default branch (in most cases this is `master`) and/or the commit that you
+  want to tag**. In almost all cases, tags should be created on the default branch for
+  automation purposes as a lot of our deliverables get pushed only when the default branch
+  has a specific tag. This checkout step is somewhat optional as `git tag` can also take
+  in a commit to tag but this procedure is less error-prone vs direct tagging.
+
+  Example of checking out the current upstream main branch history
+  ```sh-session
+  $ git fetch
+  $ git checkout master
+  $ git reset --hard origin/master
+  ```
+- **Verify that the git history looks as expected**. Tools like `git log`, `tig`, `gitk`, and
+  others can help you verify that the history is in the expected state.
+- **Create the annotated tag**. To create the tag, use the `git tag -a` command. The main tag
+  **must** be in `v<semver>` tri-dotted format:
+  ```sh-session
+  $ # If you don't have a GPG key set up
+  $ git tag -a "v1.2.3"
+
+  $ # If you do have a GPG key set up
+  $ git tag -s "v1.2.3"
+
+  $ # Make sure to add a commit header and description when the editor window opens
+  ```
+
+- **Verify that the tag was created correctly**. You can check this easily with `git describe`:
+  ```sh-session
+  $ git describe
+  v1.2.3
+  ```
+  If the version returned by `git describe` does not match your created tag, it was not correctly
+  created.
+- **Push the tag to the upstream repository**. This step makes the tag available to other users
+  of the repo:
+  ```sh-session
+  $ git push origin v1.2.3
+  ```
+
+  > Note: Overwriting or removing old tags is extremely discouraged
+
+## GitHub Releases
+
+GitHub releases are the main way that our users consume some of our end products so it is very
+important that these are maintained 1:1 along with tags.
+
+> **Note: Only full releases are considered for inclusion in the Conjur OSS Suite**
+
+### Draft Release Creation
+
+Once the [tag](#tagging) has been pushed to upstream repository and you are ready to create an
+official release of that tagged version we can go to the GitHub UI to first create a draft release:
+- **Navigate to the releases page**. In GitHub, under `<main_repo_url>/releases` you can find
+  the list of all relases that are available for the repository. We will click the `Draft a new release`
+  in the top right to get us into the "new release" page.
+- **Fill out the relevant release info**. Make sure that:
+  - `Tag version` is the tag you created for this release.
+  - `Release title` matches exactly the tag name (`v<semver>`).
+  - `Describe this release` is populated with changelog entries for this tagged version.
```

Comment:
it might be worth adding more detail here - that you should add a header `## Change log` and then copy / paste the section from the CHANGELOG.md file for this version into the release description

Reply:
Header might not be needed but I'll add the note about copy/pasting.